### PR TITLE
File download testing

### LINKED ISSUE #13378: Add support for "File download" testing in Cypress
**Is your feature request related to a problem? Please describe.**
We need to support file download testing in Cypress. Some of the affected issues: #10803, #12964 and #13271.

**Describe the solution you'd like**
There is an example in [Cypress Example Recipes](https://github.com/cypress-io/cypress-example-recipes) for [testing dom downloads](https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/testing-dom__download) along with the plugin configuration and one proposed library ([read-excel-file](https://www.npmjs.com/package/read-excel-file)).

That is a good starting point for our own Cypress configuration.

**Describe alternatives you've considered**
The main reason behind opening this issue is a discussion about which library for reading Excel files in node should we use.

During a research, I found [this topic](https://github.com/catamphetamine/read-excel-file/issues/38) in read-excel-file repo where people complained about that library struggling or failing to parse large files (and we have at least one issue that is dealing with 100k+ rows). [This comment](https://github.com/catamphetamine/read-excel-file/issues/38#issuecomment-544286628) in that very topic suggested they have solved the issue with a different library called [SheetJS](https://github.com/SheetJS/sheetjs). Even read-excel-file is officially suggesting it for large files support. This might be an overkill for our purposes, but I wanted to include it in the discussion nevertheless.

Does anyone have some other solution or direct experience with this topic?

**Additional context**
- [Cypress documentation (Change download directory)](https://docs.cypress.io/api/plugins/browser-launch-api.html#Change-download-directory)
- https://medium.com/javascript-in-plain-english/how-to-read-an-excel-file-in-node-js-6e669e9a3ce1
- [E2E Testing of excel file downloads with cypress](https://dev.to/viveknayyar/e2e-testing-of-excel-downloads-with-cypress-21fb)

**Libraries:**

|Library          | Size | Last updated | Licence |
|----------|-----------|------------|------------|
| https://www.npmjs.com/package/read-excel-file   |     1.88 MB   |      2 months ago| MIT |
| https://www.npmjs.com/package/xlsx   |     8.87 MB     |      24 days ago     | Apache-2.0 |
| https://www.npmjs.com/package/node-xlsx |     24.6 kB     |      a year ago     | Apache-2.0 |

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish?
- Adds a functionality to test (excel) file downloads
- Closes #13378 

### How to test this manually?
- `yarn test-cypress-open`
- `frontend/test/metabase/scenarios/excel/downloads.cy.spec.js`
- Tests should pass

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
5.4.0 is out already 
https://github.com/cypress-io/cypress/releases/tag/v5.4.0

File: frontend/test/metabase/scenarios/excel/downloads.cy.spec.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const xlsx = require("xlsx");
+
+describe("file download", () => {
+  before(restore);
+  beforeEach(() => {
+    signInAsAdmin();
+  });
+
+  it("downloads Excel file", () => {
```
Comment (dacort) [inline frontend/test/metabase/scenarios/excel/downloads.cy.spec.js]: Discussed in Slack, but pushing up a change to refactor this a little bit.

**Review by dacort — CHANGES_REQUESTED** [review]
We also need to test the different cases mentioned in https://github.com/metabase/metabase/issues/10803#issuecomment-638480542 as there are a number of issues with date/timestamp formats. I attempted to start this (see code below) but ran into some issues with unexpected results. Feel free to pick it up, but I can get to it tomorrow as well.

```javascript


  describe.only("for unsaved questions", () => {
    it("should format the date properly", () => {
      // Setup the query
      cy.visit("/question/new");
      cy.findByText("Native query").click();
      cy.get(".ace_editor").type(
        "SELECT PARSEDATETIME('2020-06-03', 'yyyy-MM-dd') AS \"birth_date\", PARSEDATETIME('2020-06-03 23:41:23', 'yyyy-MM-dd hh:mm:ss') AS created_at",
      );
      cy.get(".Icon-play")
        .first()
        .click();
      cy.get(".Icon-download").click();

      cy.wrap(testCases).each(testCase => {
        let downloadClassName = `.Icon-${testCase.type}`;
        let endpoint = `/api/dataset/${testCase.type}`;
        let sheetName = testCase.firstSheetName;

        cy.log(`downloading a ${testCase.type} file`);

        cy.get(downloadClassName)
          .parent()
          .parent()
          .get('input[name="query"]')
          .invoke("val")
          .then(download_query_params => {
            cy.request({
              url: endpoint,
              method: "POST",
              form: true,
              body: { query: download_query_params },
              encoding: "binary",
            }).then(resp => {
              cy.log(resp.body);
              let workbook = xlsx.read(resp.body, { type: "binary" });

              expect(workbook.SheetNames[0]).to.eq(sheetName);
              expect(workbook.Sheets[sheetName]["A1"].v).to.eq("birth_date");
              expect(workbook.Sheets[sheetName]["A2"].v).to.eq("2020-06-03");
            });
          });
      });
    });
  });
```

**Comment by dacort** [comment]
> We also need to test the different cases mentioned in [#10803 (comment)](https://github.com/metabase/metabase/issues/10803#issuecomment-638480542) as there are a number of issues with date/timestamp formats.

Merging this as the initial step in being able to further test Excel/CSV downloads.

**Review by dacort — APPROVED** [review]
🚀

**Pull request merged**